src/to-contain-html.js

```diff
@@ -0,0 +1,24 @@
+import {matcherHint, printReceived} from 'jest-matcher-utils'
+import {checkHtmlElement} from './utils'
+
+export function toContainHtml(container, htmlText) {
+  const stringToHtml = item =>
```

Comment:
You could move this out of `toContainHTML` and combine it with `checkHtmlElement`:

```js
function checkHtmlText(htmlText, ...args) {
  const htmlElement = typeof htmlText === 'string'
    ? new DOMParser().parseFromString(htmlText, 'text/html').body.firstChild
    : null
  checkHtmlElement(htmlElement, ...args)
}
```

And then use it instead of `checkHtmlElement` below:

```js
checkHtmlElement(container, toContainHtml, this)
checkHtmlText(htmlText, toContainHtml, this)
```